src/engine/sparqlExpressions/SparqlExpressionGenerators.h

```diff
@@ -71,26 +67,37 @@ CPP_template(typename T, typename Transformation = std::identity)(
 }
 
 template <typename Transformation = std::identity>
-inline cppcoro::generator<
-    const std::decay_t<std::invoke_result_t<Transformation, Id>>>
-resultGenerator(ad_utility::SetOfIntervals set, size_t targetSize,
-                Transformation transformation = {}) {
-  size_t i = 0;
-  const auto trueTransformed = transformation(Id::makeFromBool(true));
-  const auto falseTransformed = transformation(Id::makeFromBool(false));
-  for (const auto& [begin, end] : set._intervals) {
-    while (i < begin) {
-      co_yield falseTransformed;
-      ++i;
+inline auto resultGenerator(const ad_utility::SetOfIntervals& set,
+                            size_t targetSize,
+                            Transformation transformation = {}) {
+  struct Bounds {
+    size_t num_;
+    bool value_;
+  };
+  // TODO<joka921> could use inlined vector.
```

Comment:
Do it, or remove the comment. I fear that otherwise this TODO will be one of many forgotten ones